### PR TITLE
Shed fixes0216

### PR DESCRIPTION
**Pull request by coachFitz0807**
This includes some carousel fixes and tuning some of the auto paths

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
     public static final int CAROUSEL_CAN_ID = 12; // CAN ID for the spinner for carousel
     public static final double CAROUSEL_SHOOTER_SPEED = 0.8; // This is just percent output
     public static final double CAROUSEL_INTAKE_SPEED = 0.8; // This is just percent output
-    public static final int CAROUSEL_MAX_BALLS = 3;
+    public static final int CAROUSEL_MAX_BALLS = 6;
```
Comment (dwillins) [inline src/main/java/frc/robot/Constants.java:47]: max ball count should be 3 for now

File: src/main/java/frc/robot/commands/BounceAuto.java [hunk 2]
```diff
@@ -84,10 +87,10 @@ public BounceAuto(DriveTrain robotDrive, DriveCommand drivecommand) {
                 configForward);
                   
         System.out.println("DEBUG: Bounce path");
-        System.out.println("DEBUG: maxSpeed = " + maxSpeed + " maxAcceleration = " + maxAccel);
+        System.out.print("DEBUG: maxSpeed = " + maxSpeed + " maxAcceleration = " + maxAccel + " ");
         double pathTime = backTrajectory1.getTotalTimeSeconds() + backTrajectory2.getTotalTimeSeconds()
             + forwardTrajectory1.getTotalTimeSeconds() + forwardTrajectory2.getTotalTimeSeconds();
-        System.out.println("DEBUG: Bounce path time = " + pathTime);
+        System.out.println("Path time = " + pathTime);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/BounceAuto.java:93]: again. random change and not actually useful.

File: src/main/java/frc/robot/commands/BounceAuto.java [hunk 3]
```diff
@@ -36,14 +37,16 @@ public BounceAuto(DriveTrain robotDrive, DriveCommand drivecommand) {
...
 
-        double maxSpeed = 2.0;
+        double maxSpeed = 3.0;
         double maxAccel = 3.0;
 
+        TrajectoryConstraint centripetalAccelerationConstraint = new CentripetalAccelerationConstraint(2);
+
         TrajectoryConfig configForward = new TrajectoryConfig(maxSpeed, maxAccel)
-                .setKinematics(Constants.kDriveKinematics).addConstraint(autoVoltageConstraint);
+                .setKinematics(Constants.kDriveKinematics).addConstraint(autoVoltageConstraint).addConstraint(centripetalAccelerationConstraint);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/BounceAuto.java:46]: if we touch these files, these lines should split to be shorter.

File: src/main/java/frc/robot/commands/InterstellarAccuracy.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants;
+import frc.robot.FieldMapHome;
+import frc.robot.subsystems.Carousel;
+import frc.robot.subsystems.DriveTrain;
+import frc.robot.subsystems.Shooter;
+public class InterstellarAccuracy extends SequentialCommandGroup implements AutoCommandInterface {
+    Rotation2d rotation180 = Rotation2d.fromDegrees(180.0);
+    Rotation2d rotation270 = Rotation2d.fromDegrees(270.0);
+    Rotation2d rotation90 = Rotation2d.fromDegrees(90.0);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/InterstellarAccuracy.java:30]: these ought to be "final", and I believe 90 and 270 are not used?

File: src/main/java/frc/robot/commands/CarouselCommand.java [hunk 5]
```diff
@@ -35,14 +37,25 @@ public CarouselCommand(Carousel carousel) {
...
+      moveToNextSlot();
+    }
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
   
+    // Print state value to smart dashboard
+    SmartDashboard.putString("carousel/state", state.toString());
```
Comment (prensing) [inline src/main/java/frc/robot/commands/CarouselCommand.java:57]: this ought to be "Carousel" to match what is in Carousel.

File: src/main/java/frc/robot/commands/ShooterCommand.java [hunk 6]
```diff
@@ -164,12 +167,15 @@ else if (currentControlMode == ControlMethod.HOLD) {
...
     shooter.vision.setMode(VisionMode.INTAKE);
-    carousel.resetBallCount();
     carousel.spin(0.0);
+    carousel.resetBallCount();
     carouselCommand.schedule();
+    System.out.println("Shooter: carouselCommand scheduled" + carouselCommand);
     //if (rescheduleDriveCommand) {
      // driveCommand.schedule();
     //}
+  SmartDashboard.putString("shooter/Shooting", "Idle");
```
Comment (prensing) [inline src/main/java/frc/robot/commands/ShooterCommand.java:177]: bad formatting

File: src/main/java/frc/robot/subsystems/DriveTrain.java [hunk 7]
```diff
@@ -216,8 +217,8 @@ public void allDrive(double throttle, double rotate, boolean squaredInputs) {
             if (Math.abs(rotate) < 0.1) 
                 rotate = 0;
         }
-       differentialDrive.arcadeDrive(throttle, -rotate, squaredInputs);
-       // differentialDrive.curvatureDrive(throttle, -rotate, false);
+    //    differentialDrive.arcadeDrive(throttle, -rotate, squaredInputs);
+       differentialDrive.curvatureDrive(throttle, -rotate, false);
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/DriveTrain.java:221]: bad indent.

**Review by prensing — CHANGES_REQUESTED** [review]
I was a bit picky. Many of my comments can be ignored if you want to get these in. 

Not sure how the "shooter comand pseudocode" got in here....

File: src/main/java/frc/robot/commands/AutoNavPaths.java [hunk 8]
```diff
@@ -92,11 +102,11 @@ public AutoNavPaths(DriveTrain robotDrive, DriveCommand drivecommand, Path autoI
                 configForward);
 
         System.out.println("DEBUG: AutoNav path " + autoID.name());
-        System.out.println("DEBUG: maxSpeed = " + maxSpeed + " maxAcceleration = " + maxAccel);
+        System.out.print("DEBUG: maxSpeed = " + maxSpeed + " maxAcceleration = " + maxAccel + " ");
         // for (State state : backTrajectory.getStates()) {
         //     System.out.println("DEBUG: backTrajectory STATE "+ state.poseMeters);
         // }
-        System.out.println("DEBUG: AutoNav path time = " + forwardTrajectory.getTotalTimeSeconds());
+        System.out.println("Path time = " + forwardTrajectory.getTotalTimeSeconds());
```
[the remark this answers is not in the document]
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/AutoNavPaths.java:109]: There were actually three println statements, but i thought the output would be easier to read if all the data was on one line. It was easier to just turn the second println into a print

File: src/main/java/frc/robot/commands/BounceAuto.java (hunk 2 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/BounceAuto.java:93]: Same reason as before. Makes the debug output more readable.

File: src/main/java/frc/robot/commands/CarouselCommand.java (hunk 5 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/CarouselCommand.java:57]: Agree

File: src/main/java/frc/robot/commands/InterstellarAccuracy.java (hunk 4 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/InterstellarAccuracy.java:30]: Correct.

File: src/main/java/frc/robot/commands/ShooterCommand.java (hunk 6 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/ShooterCommand.java:177]: Concur

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/Constants.java:47]: Correct

File: src/main/java/frc/robot/subsystems/DriveTrain.java (hunk 7 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/subsystems/DriveTrain.java]: Concur

File: src/main/java/frc/robot/commands/AutoNavPaths.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (prensing) [reply src/main/java/frc/robot/commands/AutoNavPaths.java:109]: Well, OK. However, now if you uncomment the State printing, the formatting will be all screwed up, with the 1st state after the Accel and the time on a separate line.

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (prensing) [reply src/main/java/frc/robot/Constants.java]: Fixed.

File: src/main/java/frc/robot/Constants.java [hunk 9]
```diff
@@ -99,7 +99,7 @@
 
     public static final double ROBOT_PITCH_ANGLE_FOR_CLIMB = 5.0;
 
-    public static final int CAROUSEL_FIFTH_ROTATION_TICKS = 12561;
+    public static final int CAROUSEL_FIFTH_ROTATION_TICKS = 12615;
```
[the remark this answers is not in the document]
Comment (prensing) [reply src/main/java/frc/robot/Constants.java]: Fixed

File: src/main/java/frc/robot/commands/BounceAuto.java (hunk 3 above)
Comment (prensing) [reply src/main/java/frc/robot/commands/BounceAuto.java]: Looks good. Should do the others when they are changed, but not now.

File: src/main/java/frc/robot/commands/InterstellarAccuracy.java (hunk 4 above)
Comment (prensing) [reply src/main/java/frc/robot/commands/InterstellarAccuracy.java]: Fixed.

**Review by prensing — APPROVED** [review]

**Pull request merged**